### PR TITLE
Add documentation about the different renderers, their pros/const, and availability in backends and API

### PR DESCRIPTION
**Pull request by tronical**

### CODE REVIEW

File: docs/reference/src/advanced/backends_and_renderers.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+Slint comes with different renderers that use different techniques and libraries to turn
+your scene of elements into pixels. Slint picks a renderer backend on your choice of Backend
+as well as the features you've selected at Slint compilation time.
+
+
+### Qt Renderer
+
+The Qt renderer comes with the [Qt backend](backend_qt.md) and renders using QPainter:
+
+ - Runs anywhere.
```
Comment (ogoffart) [inline docs/reference/src/advanced/backends_and_renderers.md:54]: Contradicts with the fact that it needs Qt.

File: docs/reference/src/advanced/backends_and_renderers.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+### Software Renderer
+
+- Runs anywhere, highly portable, and lightweight.
+- Software rendering, no GPU acceleration.
+- Supports partial rendering.
+- Suitable for Microcontrollers.
+- Available in the [Winit backend](backend_winit.md).
+- Public [Rust](slint-rust:platform/software_renderer/) and [C++](slint-cpp:api/classslint_1_1platform_1_1SoftwareRenderer) API.
+
```
Comment (ogoffart) [inline docs/reference/src/advanced/backends_and_renderers.md:66]: Could also mention that it has limitations: no path, image smooth scaling, image rotation, shadow, some text stuff missing, ...

**Review by ogoffart — APPROVED** [review]

File: docs/reference/src/advanced/backends_and_renderers.md (hunk 1 above)
Comment (tronical) [reply docs/reference/src/advanced/backends_and_renderers.md:54]: Okay, what do you suggest? Remove it? (no objection, just asking for preferred course of action :)

File: docs/reference/src/advanced/backends_and_renderers.md (hunk 2 above)
Comment (tronical) [reply docs/reference/src/advanced/backends_and_renderers.md:66]: Good point, I'll add that.

File: docs/reference/src/advanced/backends_and_renderers.md (hunk 1 above)
Comment (ogoffart) [reply docs/reference/src/advanced/backends_and_renderers.md:54]: I think I'd leave it out

**Review by ogoffart — APPROVED** [review]

File: docs/reference/src/advanced/backends_and_renderers.md (hunk 1 above)
Comment (tronical) [reply docs/reference/src/advanced/backends_and_renderers.md]: Done :)

**Pull request merged**